PIMbench/cpp-knn/baselines/CPU/knn.cpp

```diff
@@ -0,0 +1,280 @@
+#include <stdint.h>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <cmath>
+#include <unordered_map>
+#include <chrono>
+#include <vector>
+#include <string>
+#include <queue>
+#include <iomanip>
+using namespace std;
+
+#include <omp.h>
+#include "../../../utilBaselines.h"
+#include <cfloat>
+
+vector<vector<int>> dataPoints;
+vector<vector<int>> testPoints;
+
+// Params ---------------------------------------------------------------------
+typedef struct Params
+{
+    int numTestPoints;
+    int numDataPoints;
+    int dimension;
+    int k;
+    int numThreads;
+    char *inputTestFile;
+    char *inputDataFile;
+    int target;
+} Params;
+
+void usage()
+{
+    fprintf(stderr,
+            "\nUsage:  ./program [options]"
+            "\n"
+            "\n    -t    # of threads (default=8)"
+            "\n    -n    number of reference points (default=65536 points)"
+            "\n    -m    number of query points (default=100 points)"
+            "\n    -d    dimension (default=2)"
+            "\n    -k    value of K (default=20)"
+            "\n    -x    target dimension index of the data set(default=1)"
+            "\n    -i    input file containing training datapoints (default=generates datapoints with random numbers)"
+            "\n    -j    input file containing testing datapoints (default=generates datapoints with random numbers)"
+            "\n");
+}
+
+struct Params input_params(int argc, char **argv)
+{
+    struct Params p;
+    p.numDataPoints = 65536;
+    p.numTestPoints = 100;
+    p.dimension = 2;
+    p.k = 20;
+    p.numThreads = 8;
+    p.target = 1;
+    p.inputTestFile = nullptr;
+    p.inputDataFile = nullptr;
+
+    int opt;
+    while ((opt = getopt(argc, argv, "h:k:t:n:m:d:x:i:j:")) >= 0)
+    {
+        switch (opt)
+        {
+        case 'h':
+            usage();
+            exit(0);
+            break;
+        case 'n':
+            p.numDataPoints = atoll(optarg);
+            break;
+        case 'm':
+            p.numTestPoints = atoll(optarg);
+            break;
+        case 'd':
+            p.dimension = atoll(optarg);
+            p.target = p.dimension - 1;
+            break;
+        case 'k':
+            p.k = atoi(optarg);
+            break;
+        case 't':
+            p.numThreads = atoi(optarg);
+            break;
+        case 'x':
+            p.target = atoi(optarg);
+            break;
+        case 'i':
+            p.inputDataFile = optarg;
+            break;
+        case 'j':
+            p.inputTestFile = optarg;
+            break;
+        default:
+            fprintf(stderr, "\nUnrecognized option!\n");
+            usage();
+            exit(0);
+        }
+    }
+
+    return p;
+}
+
+inline int calculateDistance(const vector<int> &pointA, const vector<int> &pointB, int dim, int target) {
+    int sum = 0;
+    for (int i = 0; i < dim; i++)
+    {
+        if (i == target) {continue;}
+        sum += abs(pointA[i] - pointB[i]);
```

Comment:
```suggestion
        if (i != target)
           sum += abs(pointA[i] - pointB[i]);
```
Please verify this before updating